### PR TITLE
Clarified span sending on AWS lambda

### PR DESCRIPTION
**Pull request by kirrg001**
refs https://jsw.ibm.com/browse/INSTA-10320

Please see comments in the ticket.
PR contains two commits.

### CODE REVIEW

File: packages/aws-lambda/test/multiple_data/test.js [hunk 1]
```diff
@@ -26,72 +26,133 @@ const qualifiedArn = `${unqualifiedArn}:${version}`;
...
+
+    before(async () => {
+      control = new Control({
+        faasRuntimePath: path.join(__dirname, '../runtime_mock'),
+        handlerDefinitionPath: path.join(__dirname, './lambda'),
+        startBackend: true,
+        env: {
+          INSTANA_AGENT_KEY: instanaAgentKey,
+          WITH_CONFIG: 'true',
+          INSTANA_NUMBER_OF_ITERATIONS: 100
```
Comment (kirrg001) [inline packages/aws-lambda/test/multiple_data/test.js:109]: Here is the new test case: 100 test requests.

File: packages/aws-lambda/test/multiple_data/test.js [hunk 2]
```diff
@@ -26,72 +26,133 @@ const qualifiedArn = `${unqualifiedArn}:${version}`;
...
+          return retry(
+            () => {
+              return Promise.all([
+                //
+                control.getSpans(),
+                control.getRawBundles()
+              ]).then(([spans, rawBundles]) => {
+                // one bundle request with 101 spans
+                expect(rawBundles.length).to.equal(1);
+                expect(spans.length).to.equal(101);
```
Comment (kirrg001) [inline packages/aws-lambda/test/multiple_data/test.js:145]: One bundle request containing 101 spans.

Comment (kirrg001) [reply packages/aws-lambda/test/multiple_data/test.js:145]: I will take a look if there is a quick fix we can apply (not part of this PR).

**Comment by kirrg001** [comment]
We might need to change the default buffering behavior for AWS Lambda (long-term).
Short-term: Enable batching.

File: packages/aws-lambda/src/wrapper.js [hunk 3]
```diff
@@ -432,9 +433,9 @@ function postHandler(entrySpan, error, result, callback) {
   const spans = spanBuffer.getAndResetSpans();
 
   // We want that all upcoming spans are send immediately to the BE.
-  // Span collection happens all the time and they are buffered via spanBuffer.
-  // The spanBuffer will only send out the spans early if the payload becomes too big.
-  // When the Lambda handler finishes we trigger `sendBundle`.
+  // Span collection happens all the time, but for AWS Lambda sending spans early via spanBuffer
```
Comment (aryamohanan) [inline packages/aws-lambda/src/wrapper.js:436]: Qs: For AWS Lambda we are not using spanBuffer?

Comment (kirrg001) [reply packages/aws-lambda/src/wrapper.js:436]: Exactly. The comment describes why.

Comment (aryamohanan) [reply packages/aws-lambda/src/wrapper.js:436]: Great! I'm glad you found this and updated the comments ❤️

File: packages/serverless/src/backend_connector.js [hunk 4]
```diff
@@ -117,6 +117,10 @@ exports.setLogger = function setLogger(_logger) {
  * the variable is always true.
  */
 exports.sendBundle = function sendBundle(bundle, finalLambdaRequest, callback) {
+  logger.debug(
+    `Sending bundle to Instana (no. of spans: ${(bundle && bundle.spans && bundle.spans.length) || 'unknown'})`
```
Comment (abhilash-sivan) [inline packages/serverless/src/backend_connector.js:121]: s: can we use optional chaining here? IMO that might be an easy syntax
`Sending bundle to Instana (no. of spans: ${bundle?.spans?.length ?? 'unknown'})`

Comment (kirrg001) [reply packages/serverless/src/backend_connector.js:121]: Yes thank you. Will change!

**Review by abhilash-sivan — APPROVED** [review]

**Pull request merged**